wallet-new/integration/TransactionSpecs.hs

```diff
@@ -47,6 +56,72 @@ transactionSpecs wRef wc = do
 
             map txId resp `shouldContain` [txId txn]
 
+        it "asset-locked wallets can receive funds and transaction shows in index" $ do
+            genesis <- genesisWallet wc
+            (fromAcct, _) <- firstAccountAndId wc genesis
+
+            wallet <- genesisAssetLockedWallet wc
+            (toAcct, toAddr) <- firstAccountAndId wc wallet
+
+            let payment = Payment
+                    { pmtSource =  PaymentSource
+                        { psWalletId = walId genesis
+                        , psAccountIndex = accIndex fromAcct
+                        }
+                    , pmtDestinations = pure PaymentDistribution
+                        { pdAddress = addrId toAddr
+                        , pdAmount = tenthOf (accAmount fromAcct)
+                        }
+                    , pmtGroupingPolicy = Nothing
+                    , pmtSpendingPassword = Nothing
+                    }
+                tenthOf (V1 c) = V1 (Core.mkCoin (Core.getCoin c `div` 10))
+
+            etxn <- postTransaction wc payment
+
+            txn <- fmap wrData etxn `shouldPrism` _Right
+
+            eresp <- getTransactionIndex wc (Just (walId wallet)) (Just (accIndex toAcct)) Nothing
+            resp <- fmap wrData eresp `shouldPrism` _Right
+
+            map txId resp `shouldContain` [txId txn]
+
+        it "sending from asset-locked address in wallet with no ther addresses gets 0 confirmations from core nodes" $ do
+            genesis <- genesisAssetLockedWallet wc
+            (fromAcct, _) <- firstAccountAndId wc genesis
+
+            wallet <- sampleWallet wRef wc
+            (toAcct, toAddr) <- firstAccountAndId wc wallet
+
+            let payment = Payment
+                    { pmtSource =  PaymentSource
+                        { psWalletId = walId genesis
+                        , psAccountIndex = accIndex fromAcct
+                        }
+                    , pmtDestinations = pure PaymentDistribution
+                        { pdAddress = addrId toAddr
+                        , pdAmount = tenthOf (accAmount fromAcct)
+                        }
+                    , pmtGroupingPolicy = Nothing
+                    , pmtSpendingPassword = Nothing
+                    }
+                tenthOf (V1 c) = V1 (Core.mkCoin (Core.getCoin c `div` 10))
+
+            etxn <- postTransaction wc payment
+
+            txn <- fmap wrData etxn `shouldPrism` _Right
+
+            threadDelay 10000
+            eresp <- getTransactionIndex wc (Just (walId wallet)) (Just (accIndex toAcct)) Nothing
+            resp <- fmap wrData eresp `shouldPrism` _Right :: IO [ Transaction]
+            let txnEntry = head (filter ( \x -> (txId x) == (txId txn)) resp)
+            log $ "Resp   : " <> ppShowT txnEntry
+            case txnEntry of
+                Just txnEntry' -> do
+                    txConfirmations txnEntry' `shouldBe` 0
+                Nothing -> do
+                    1 `shouldBe` (0 :: Integer)
```

Comment:
maybe switch to https://hackage.haskell.org/package/base-4.11.1.0/docs/Prelude.html#v:fail